libcomposefs/lcfs-writer.c

```diff
@@ -390,12 +390,6 @@ int lcfs_write_to(struct lcfs_node_s *root, struct lcfs_write_options_s *options
 	struct lcfs_ctx_s *ctx;
 	int res;
 
-	// Verify the root; because lcfs_node_add_child also verifies children,
-	// we should have sanity checked all nodes.
-	if (lcfs_node_last_ditch_validation(root) < 0) {
```

Comment:
One very minor nag: we handle validation of the node from the main writer code currently but your patch moves it to the erofs implementation code.  Of course, there *are* no other implementations... but it seems a bit odd.

I guess the implementation is the thing iterating the nodes in the end, though, so there's not really much of another way...